runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -4,12 +4,30 @@ local Range = require('vim.treesitter._range')
 
 local ns = api.nvim_create_namespace('treesitter/highlighter')
 
----@alias vim.treesitter.highlighter.Iter fun(end_line: integer|nil): integer, TSNode, vim.treesitter.query.TSMetadata, TSQueryMatch
+--- @class vim.treesitter.highlighter.Info
+--- @inlinedoc
+--- The range of the highlight
+--- @field range Range6
+--- The group of the highlight
+--- @field hl_group string
+--- The priority of the highlight
+--- @field priority integer
+--- The conceal of the highlight, nil for no conceal
+--- @field conceal string?
+--- Whether the highlight is spell or not
+---   - If `nil` then the highlight doesn't have a spell set
+---   - If `true` then the highlight is spell
+---   - If `false` then the highlight is nospell
```

Comment:
is this referring to 'nospell' option or something else (extmark-specific, or syntax specific...?)

Reply:
Note that `@spell` (and `@conceal`) is not technically a highlight. I would leave it out for now.

Reply:
> Note that `@spell` (and `@conceal`) is not technically a highlight. I would leave it out for now.

Is there some other way to get that info? Maybe we should rename the function if "highlight" is objectionable. What matters is that it gets useful info, the technical distinction seems minor?

Reply:
Well, the question then becomes (again) "what is this function (meant to be) used for?"

Reply:
Could we consider making this internal only for now in order to progress this easier?

Reply:
Sure